src/components/accounts/SetupImplicit.js

```diff
@@ -134,167 +107,143 @@ class SetupImplicit extends Component {
+                            Mixpanel.track("CA Check balance from implicit: insufficient")
+                        }
+                    },
+                    (e) => { 
+                        if (e.message.indexOf('exist while viewing') === -1) {
+                            throw e
+                        }
+                        this.setState({ balance: 0 })
+                    }
+                )
+            }
         }
-        )
     }
 
     componentDidMount = () => {
-
-        this.connection = nearApiJs.Connection.fromConfig({
-            networkId: NETWORK_ID,
-            provider: { type: 'JsonRpcProvider', args: { url: NODE_URL + '/' } },
-            signer: {},
-        })
-
-        // TODO: Use wallet/Redux for queries? Or at least same connection.
-
-        clearInterval(pollingInterval)
-        pollingInterval = setInterval(this.checkBalance, 2000)
-
+        // TODO: Check if account has already been created and if so, navigate to dashboard
+        this.interval = setInterval(() => this.checkBalance(), 2000)
         this.checkMoonPay()
     }
 
     componentWillUnmount = () => {
-        clearInterval(pollingInterval)
+        clearInterval(this.interval)
     }
 
-    // TODO: Refactor: Extract utility to copy text
-    // optionally pass in string to copy: textToCopy
-    handleCopyPhrase = (textToCopy) => {
-        Mixpanel.track("CA Copy funding address")
-        if (typeof textToCopy !== 'string') {
-            textToCopy = null
-        }
-        if (navigator.share && isMobile()) {
-            navigator.share({
-                text: textToCopy
-            }).catch(err => {
-                console.log(err.message);
-            });
-        } else {
-            this.handleCopyDesktop(textToCopy);
+    handleClaimAccount = () => {
+        const { dispatch, accountId, activeAccountId } = this.props;
+        if (accountId === activeAccountId) {
+            dispatch(redirectTo('/'))
+            return;
         }
-    }
-
-    handleCopyDesktop = (textToCopy) => {
-        // TODO: Use actual textToCopy passed as parameter
-        copyText(document.getElementById('implicit-account-id'));
-        this.setState({ successSnackbar: true }, () => {
-            setTimeout(() => {
-                this.setState({ successSnackbar: false });
-            }, snackbarDuration)
-        });
+        this.setState({ claimMyAccount: true });
     }
 
     render() {
         const {
-            snackBarMessage,
-            successSnackbar,
             whereToBuy,
-            checked,
             createAccount,
             moonpayAvailable,
-            moonpaySignedURL
+            moonpaySignedURL,
+            balance,
+            claimMyAccount,
+            creatingAccount
         } = this.state
 
-        const { implicitAccountId, accountId, mainLoader } = this.props
+        const { implicitAccountId, accountId, mainLoader } = this.props;
+
+        if (createAccount) {
```

Comment:
Not a blocker, but it feels like we actually have two-components-in-one here -- each with different purposes